tests/test-firefox/test.firefox.js

```diff
@@ -480,29 +481,53 @@ describe('firefox', () => {
         });
     });
 
-    it('throws an error when the port is occupied', () => {
-      // TODO: add a retry for occupied ports.
-      // https://github.com/mozilla/web-ext/issues/283
-      const client = fake(RemoteFirefox.prototype);
-      const connectToFirefox = sinon.spy(() => Promise.resolve(client));
-      return findRemotePort({connectToFirefox})
-        .then(makeSureItFails())
-        .catch(onlyInstancesOf(WebExtError, (error) => {
-          assert.match(error.message, /Cannot listen on port/);
-          assert.equal(client.disconnect.called, true);
+    it('returns a port on first try', () => {
+      const connectToFirefox = sinon.spy(() => new Promise(
+        (resolve, reject) => {
+          reject(
+            new TCPConnectError('first call connection fails - port is free')
+          );
         }));
+      return findRemotePort({connectToFirefox})
+      .then((port) => {
```

Comment:
this block should be indented to show that it's an extension of the return statement